# App Store and Google Play In-App Profits

This is a data analysis portfolio project I worked through using DataQuest. The goal of this project is to analyze data in which helps developers understand what kind of Android and iOS apps are profitable and which attract more users on Google Play and the App Store. In this project, we will only work with apps that are free to download and install.

Since we're only working with apps that are free to download and install, it is important to focus on *in-app ads* as our main source of revenue. In other words, we want to determine how many users engage with the ads. The more users engage with the ads, the better for profit.

## Understanding and Exploring our Data

As of September 2018, there were approximately 2 million iOS apps available on the App Store, and 2.1 million Android apps on Google Play.
*Source: [Statista](https://www.statista.com/statistics/276623/number-of-apps-available-in-leading-app-stores/)*

Collecting data for over 4 million apps would be *way too much* to take on for this project! So, let's focus on a sample of the data. We're going to use these two datasets:
* [This dataset](https://www.kaggle.com/lava18/google-play-store-apps) contains approximately 10,000 Android apps from Google Play. The data was collected in August 2018. You can download the data set directly [here](https://dq-content.s3.amazonaws.com/350/googleplaystore.csv).
* [This dataset](https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps) contains approximately 7,000 iOS apps from the App Store. The data was collected in July 2017. You can download the data set directly [here](https://dq-content.s3.amazonaws.com/350/AppleStore.csv).

***

Let's start opening and exploring the two data sets we're working with.

In [1]:
from csv import reader

apple_dataset = 'AppleStore.csv'
apple_file = open(apple_dataset)
read_apple = reader(apple_file)
ios_data = list(read_apple)
ios_header = ios_data[0]
ios_data = ios_data[1:]

android_dataset = 'googleplaystore.csv'
android_file = open(android_dataset)
read_android = reader(android_file)
android_data = list(read_android)
android_header = android_data[0]
android_data = android_data[1:]

In [2]:
def explore_data(dataset, start, end, rows_columns=False):
    dataset_slice = dataset[start:end]
    for i in dataset_slice:
        print(i)
        print('\n')
    if rows_columns:
        print('Number of rows:', len(dataset))
        print('Number of columns:', len(dataset[0]))

Let's explore the Apple dataset first.

In [3]:
## Explore the Apple data set
print(ios_header)
print('\n')
explore_data(ios_data, 0, 5, True)

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


['420009108', 'Temple Run', '65921024', 'USD', '0.0', '1724546', '3842', '4.5', '4.0', '1.6.2', '9+', 'Games', '40', '5', '1', '1']


['284035177', 'Pandora - Music & Radio', '130242560', 'USD', '0.0', '1126879', '3594', '4.0', '4.5', '8.4.1', '12+', 'Music', '37', '4', '1', '1']


Number of rows: 7197
Number of columns: 16


We should identify which columns would help us with our analysis. There are 7197 apps and 16 columns. The columns that might be useful are:

* `track_name` to identify what the app name is.
* `currency` to identify which type of currency the app is.
* `price` to identify the price of the app.
* `rating_count_tot` to identify the total number of ratings based on **all versions** of the app.
* `rating_count_ver` to identify the total number of ratings based on the **current version** of the app.
* `prime_genre` to identify the genre of the app.

Now let's explore the Android dataset.

In [4]:
## Explore the Android data set
print(android_header)
print('\n')
explore_data(android_data, 0, 5, True)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


['Pixel Draw - Number Art Coloring Book', 'ART_AND_DESIGN', '4.3', '967', '2.8M', '100,000+', 'Free', '0', 'Eve

In the Android dataset, there are 10841 apps and 13 columns. The columns that might be useful are:

* `App` to identify the name of the app.
* `Category` to identify the category the app belongs to.
* `Reviews` to identify the total number of user reviews for the app.
* `Installs` to identify the total number of user installs for the app.
* `Type` to identify if the app is paid or free.
* `Price` to identify the price of the app.
* `Genres` to identify the genre of the app.

***

## Data Cleaning

Next, we need to do **data cleaning** before our analysis. This involves removing or correcting any inaccurate/duplicate data.

The Google Play dataset has a dedicated [discussion section](https://www.kaggle.com/lava18/google-play-store-apps/discussion). In the discussion section, [one of the discussions](https://www.kaggle.com/lava18/google-play-store-apps/discussion/66015) mentions an error for a certain row.

### Inaccurate Data

According to the discussion section, this entry has a missing column. The maximum app rating is 5, so since the rating for row 10472 is 19, this is inaccurate. So, let's delete this entry.

In [5]:
del(android_data[10472])
print(len(android_data))

10840


### Duplicate Entries

Now, let's take a look at any duplicate entries. According to the Google Play dataset [discussions section](https://www.kaggle.com/lava18/google-play-store-apps/discussion), some apps have duplicate entries. So, we should see how many times a duplicate app appears.

In [6]:
duplicates = []
uniques = []

for a in android_data:
    a_name = a[0]
    if a_name in uniques:
        duplicates.append(a_name)
    else:
        uniques.append(a_name)

print('Number of duplicate apps: {}'.format(len(duplicates)))
print('\n')
print('Examples of duplicate apps:', duplicates[:15])

Number of duplicate apps: 1181


Examples of duplicate apps: ['Quick PDF Scanner + OCR FREE', 'Box', 'Google My Business', 'ZOOM Cloud Meetings', 'join.me - Simple Meetings', 'Box', 'Zenefits', 'Google Ads', 'Google My Business', 'Slack', 'FreshBooks Classic', 'Insightly CRM', 'QuickBooks Accounting: Invoicing & Expenses', 'HipChat - Chat Built for Teams', 'Xero Accounting Software']


Now that we've counted and confirmed the number of duplicate apps, we need to remove these entries. However, we shouldn't remove them arbitrarily. 

Since there are a different number of total user reviews per entry, it would be helpful to use the most *recent* entry. In other words, the higher the number of total user reviews is, the more recent the data is.

In [7]:
max_reviews = {}

for a in android_data:
    a_name = a[0]
    a_reviews = float(a[3])
    
    if a_name in max_reviews and max_reviews[a_name] < a_reviews:
        max_reviews[a_name] = a_reviews

    elif a_name not in max_reviews:
        max_reviews[a_name] = a_reviews

So, the dictionary `max_reviews` will contain **unique keys**, or unique apps. This is because the above code loops through the Android dataset and adds entries to the dictionary with each key as an app name, and each corresponding value as the app's highest number of user reviews. Since each value is the app's highest number of user reviews, each entry will be unique, as desired.

We know that there are 1181 duplicates. After we remove the duplicates, we should be left with 9659 rows. Why? Because the number of apps in our Android dataset is 10840. Hence, 10840 - 1181 = 9659.

In [8]:
print('Expected:', len(android_data) - 1181)
print('Actual: ', len(max_reviews))

Expected: 9659
Actual:  9659


Let's use the dictionary to create our cleaned Android dataset without any duplicates. In order to do this, we need to:

* Create two lists. We'll create one list, `android_clean` and another list, `android_added`. 

* Loop through the Android dataset and pull each app name and the total number of user reviews.

* If the total number of user reviews of the pulled app matches the total number of user reviews of the app in `max_reviews`, then that must be the highest number of user reviews for the pulled app! If we're going to go ahead and add it to `android_clean`, then we also need to make sure it isn't already in `android_added`. 

* The reason why we need to make sure it isn't already in `android_added` is because it's possible that multiple entries have the *same* total number of user reviews. If we didn't have this condition, we'd risk having more duplicates.

In [9]:
android_clean = []
android_added = []

for a in android_data:
    a_name = a[0]
    a_reviews = float(a[3])
    
    if (max_reviews[a_name] == a_reviews) and (a_name not in android_added):
        android_clean.append(a)
        android_added.append(a_name)

Let's explore the new data set to confirm our cleaned dataset has the correct number of rows.

In [10]:
explore_data(android_clean, 0, 3, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 9659
Number of columns: 13


We have 9659 rows, as desired. In the App Store dataset, there are no duplicates.

### Non-English Apps

Next, since we're focusing on our apps being geared towards an English-speaking users in this project, we're going to need to only analyze apps that are for English-speaking users. 

In both data sets, there are apps with names that likely weren't designed for English-speaking users. See examples below.

In [11]:
## Apple data ##
print(ios_data[6731][1])
print('\n')

## Android data ##
print(android_clean[4412][0])

【脱出ゲーム】絶対に最後までプレイしないで 〜謎解き＆ブロックパズル〜


中国語 AQリスニング


Since these apps aren't useful for our analysis, it's best to just remove them. We can start by seeing which app names contain a non-English character.

Using the `ord()` function will allow us to see a corresponding number for each character.

In [12]:
print(ord('a'))
print(ord('最'))

97
26368


You can see the corresponding number for "最" is 26368.

According to the [ASCII](https://en.wikipedia.org/wiki/ASCII), the numbers corresponding to characters in English text are within the range of 0 to 127. Therefore, we can assume that if a character is non-English, it would have a value greater than 127.

We can use this to determine whether an app name contains a non-English character.

In [13]:
def eng_char(string):
    for char in string:
        if ord(char) > 127:
            return False
    return True

print(eng_char('Facebook'))
print(eng_char('Instagram'))
print(eng_char('中国語 AQリスニング'))
print(eng_char('Docs To Go™ Free Office Suite'))
print(eng_char('Instachat 😜'))

True
True
False
False
False


There's one issue with using the above function even though it technically *can* correctly determine whether a character is English or not. 

In some English app names, there are uses of emojis or symbols that may exceed the ASCII range. We'd be significantly losing data if we removed every app that contained an emoji or symbol.

In order to prevent a significant loss of data, we should only remove an app if its name has more than three characters with corresponding numbers that exceed the ASCII range. 

In [14]:
def eng_char(string):
    non_eng = 0
    for char in string:
        if ord(char) > 127:
            non_eng += 1
    if non_eng > 3:
        return False
    else:
        return True

print(eng_char('Docs To Go™ Free Office Suite'))
print(eng_char('Instachat 😜'))

True
True


That should be good enough. Now, let's use this function to filter out non-English apps from both datasets.

In [15]:
android_eng = []
ios_eng = []

for a in android_clean:
    a_name = a[0]
    if eng_char(a_name):
        android_eng.append(a)
    
for a in ios_data:
    a_name = a[1]
    if eng_char(a_name):
        ios_eng.append(a)

Let's explore the data.

In [16]:
explore_data(android_eng, 0, 3, True)
print('\n')
explore_data(ios_eng, 0, 3, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 9614
Number of columns: 13


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 

After filtering out non-English apps, we have 9614 Android apps and 6183 iOS apps

So far, we were able to successfully clean the data in order to:

* Remove inaccurate data
* Remove duplicate entries
* Remove non-English apps

We have one more step in the data cleaning process.

### Isolating Free Apps

As mentioned earlier, we're only focusing on apps that are *free* to download and install. So, we need to clean our datasets a bit more since they both contain free apps and non-free apps.

In [17]:
android_new = []
ios_new = []

for a in android_eng:
    a_price = a[7]
    if a_price == '0':
        android_new.append(a)

for a in ios_eng:
    a_price = a[4]
    if a_price == '0.0' or a_price == '0':
        ios_new.append(a)

print('Total android apps: {}'.format(len(android_new)))
print('Total iOS apps: {}'.format(len(ios_new)))

Total android apps: 8864
Total iOS apps: 3222


After cleaning the data, we have a total of 8864 Android apps and 3222 iOS apps remaining.
***

## Validation Strategy

Now that we've went through cleaning the data, we can refocus on our goal. Ultimately, we want to determine which kinds of apps are likely to attract users for our source of revenue.

To minimize and prevent risk, we ideally want a validation strategy for an app:

1. Build a minimal Android version of the app, and add it to Google Play.
2. If the app has a good response from users, develop it further.
3. If the app is profitable after six months, build an iOS version of the app and add it to the App Store.

Since our end goal is to successfully add a profitable app on Google Play and the App Store, we want to find app profiles that are successful in *both* markets. 

### Most Common Apps by Genre

Let's start by determining the most common genres for each market. We need to build frequency tables for the relevant columns in our datasets. Since we're concentrating on the most common genres for each market, it might be helpful to use the `Genres` and `Category` columns from the Android dataset, and `prime_genre` from the App Store dataset.

In [18]:
def freq_table(dataset, index):
    table_dict = {}
    running_total = 0
    
    for i in dataset:
        running_total += 1
        val = i[index]
        if val in table_dict:
            table_dict[val] += 1
        else:
            table_dict[val] = 1
            
    table_percents = {}
    for key in table_dict:
        percent = (table_dict[key] / running_total) * 100
        table_percents[key] = percent
        
    return table_percents

def display_table(dataset, index):
    table = freq_table(dataset, index)
    table_display = []
    for key in table:
        key_val_as_tuple = (table[key], key)
        table_display.append(key_val_as_tuple)
        
    table_sorted = sorted(table_display, reverse = True)
    for entry in table_sorted:
        print(entry[1], ':', entry[0])

In [19]:
## iOS Genre
display_table(ios_new, 11)

Games : 58.16263190564867
Entertainment : 7.883302296710118
Photo & Video : 4.9658597144630665
Education : 3.662321539416512
Social Networking : 3.2898820608317814
Shopping : 2.60707635009311
Utilities : 2.5139664804469275
Sports : 2.1415270018621975
Music : 2.0484171322160147
Health & Fitness : 2.0173805090006205
Productivity : 1.7380509000620732
Lifestyle : 1.5828677839851024
News : 1.3345747982619491
Travel : 1.2414649286157666
Finance : 1.1173184357541899
Weather : 0.8690254500310366
Food & Drink : 0.8069522036002483
Reference : 0.5586592178770949
Business : 0.5276225946617008
Book : 0.4345127250155183
Navigation : 0.186219739292365
Medical : 0.186219739292365
Catalogs : 0.12414649286157665


Among free English iOS apps, more than half (58.16%) are games. The most common genre is gaming.

The next most common genre would be Entertainment at about 8%. Photo & video apps are just underneath Entertainment at about 5%. 

The general impression is that most apps are designed for gaming and entertainment purposes (games, photo and video, social networking) as opposed to practical purposes (education, shopping, utilities, productivity). 

However, we cannot recommend an app profile based on this. If there is a large number of apps for a particular genre, this does not necessarily mean it has a large number of users. Let's take a look at the frequency table for the Google Play dataset.

In [20]:
## Android Category
display_table(android_new, 1)

FAMILY : 18.907942238267147
GAME : 9.724729241877256
TOOLS : 8.461191335740072
BUSINESS : 4.591606498194946
LIFESTYLE : 3.9034296028880866
PRODUCTIVITY : 3.892148014440433
FINANCE : 3.7003610108303246
MEDICAL : 3.531137184115524
SPORTS : 3.395758122743682
PERSONALIZATION : 3.3167870036101084
COMMUNICATION : 3.2378158844765346
HEALTH_AND_FITNESS : 3.0798736462093865
PHOTOGRAPHY : 2.944494584837545
NEWS_AND_MAGAZINES : 2.7978339350180503
SOCIAL : 2.6624548736462095
TRAVEL_AND_LOCAL : 2.33528880866426
SHOPPING : 2.2450361010830324
BOOKS_AND_REFERENCE : 2.1435018050541514
DATING : 1.861462093862816
VIDEO_PLAYERS : 1.7937725631768955
MAPS_AND_NAVIGATION : 1.3989169675090252
FOOD_AND_DRINK : 1.2409747292418771
EDUCATION : 1.1620036101083033
ENTERTAINMENT : 0.9589350180505415
LIBRARIES_AND_DEMO : 0.9363718411552346
AUTO_AND_VEHICLES : 0.9250902527075812
HOUSE_AND_HOME : 0.8235559566787004
WEATHER : 0.8009927797833934
EVENTS : 0.7107400722021661
PARENTING : 0.6543321299638989
ART_AND_DESIGN : 

This is pretty different than what we've seen from our App Store frequency table. Family is the most common genre at about 19%. Unlike the App Store, a large number of apps are designed for productivity purposes (tools, business, lifestyle, productivity, finance, medical).

Let's take a deeper look by checking out the `Genres` column.

In [21]:
## Android Genres
display_table(android_new, -4)

Tools : 8.449909747292418
Entertainment : 6.069494584837545
Education : 5.347472924187725
Business : 4.591606498194946
Productivity : 3.892148014440433
Lifestyle : 3.892148014440433
Finance : 3.7003610108303246
Medical : 3.531137184115524
Sports : 3.463447653429603
Personalization : 3.3167870036101084
Communication : 3.2378158844765346
Action : 3.1024368231046933
Health & Fitness : 3.0798736462093865
Photography : 2.944494584837545
News & Magazines : 2.7978339350180503
Social : 2.6624548736462095
Travel & Local : 2.3240072202166067
Shopping : 2.2450361010830324
Books & Reference : 2.1435018050541514
Simulation : 2.0419675090252705
Dating : 1.861462093862816
Arcade : 1.8501805054151623
Video Players & Editors : 1.7712093862815883
Casual : 1.7599277978339352
Maps & Navigation : 1.3989169675090252
Food & Drink : 1.2409747292418771
Puzzle : 1.128158844765343
Racing : 0.9927797833935018
Role Playing : 0.9363718411552346
Libraries & Demo : 0.9363718411552346
Auto & Vehicles : 0.9250902527075

The number of categories under `Genres` can be a bit overwhelming and too granular, so it's best to focus on the `Category` column.

Overall, we've found that the App Store contains apps that are designed for fun whereas Google Play has a bit more productivity in its store.

### Most Popular Apps by Genre

Next, let's check out which kinds of apps has the most users. We can calculate the average number of installs for each app genre.

For the Google Play data set, we'll use the `Installs` column and for the App Store data set, we'll use the `rating_count_tot` since there is no installs information.

#### App Store

In [22]:
ios_genres = freq_table(ios_new, 11)


for i in ios_genres:
    total = 0
    genre_len = 0
    for a in ios_new:
        app_genre = a[11]
        if app_genre == i:
            a_ratings = float(a[5])
            total += a_ratings
            genre_len += 1
    avg_a_rat = total / genre_len
    print(i, ': {}'.format(avg_a_rat))

Social Networking : 71548.34905660378
Photo & Video : 28441.54375
Games : 22788.6696905016
Music : 57326.530303030304
Reference : 74942.11111111111
Health & Fitness : 23298.015384615384
Weather : 52279.892857142855
Utilities : 18684.456790123455
Travel : 28243.8
Shopping : 26919.690476190477
News : 21248.023255813954
Navigation : 86090.33333333333
Lifestyle : 16485.764705882353
Entertainment : 14029.830708661417
Food & Drink : 33333.92307692308
Sports : 23008.898550724636
Book : 39758.5
Finance : 31467.944444444445
Education : 7003.983050847458
Productivity : 21028.410714285714
Business : 7491.117647058823
Catalogs : 4004.0
Medical : 612.0


Navigation has the highest number of user reviews on average. Let's dig a little deeper into this.

In [23]:
for a in ios_new:
    if a[11] == 'Navigation':
        print(a[1], ':', a[5])

Waze - GPS Navigation, Maps & Real-time Traffic : 345046
Google Maps - Navigation & Transit : 154911
Geocaching® : 12811
CoPilot GPS – Car Navigation & Offline Maps : 3582
ImmobilienScout24: Real Estate Search in Germany : 187
Railway Route Search : 5


It's important to point out the two dominating apps here. Waze and Google Maps both significantly drown out the other apps in terms of user reviews. This has a large influence on why the average number of user reviews for Navigation apps is so high.

We can see that References has the second highest number of user reviews on average. Let's dig a little deeper into this one.

In [24]:
for a in ios_new:
    if a[11] == 'Reference':
        print(a[1], ':', a[5])

Bible : 985920
Dictionary.com Dictionary & Thesaurus : 200047
Dictionary.com Dictionary & Thesaurus for iPad : 54175
Google Translate : 26786
Muslim Pro: Ramadan 2017 Prayer Times, Azan, Quran : 18418
New Furniture Mods - Pocket Wiki & Game Tools for Minecraft PC Edition : 17588
Merriam-Webster Dictionary : 16849
Night Sky : 12122
City Maps for Minecraft PE - The Best Maps for Minecraft Pocket Edition (MCPE) : 8535
LUCKY BLOCK MOD ™ for Minecraft PC Edition - The Best Pocket Wiki & Mods Installer Tools : 4693
GUNS MODS for Minecraft PC Edition - Mods Tools : 1497
Guides for Pokémon GO - Pokemon GO News and Cheats : 826
WWDC : 762
Horror Maps for Minecraft PE - Download The Scariest Maps for Minecraft Pocket Edition (MCPE) Free : 718
VPN Express : 14
Real Bike Traffic Rider Virtual Reality Glasses : 8
教えて!goo : 0
Jishokun-Japanese English Dictionary & Translator : 0


We can see a similar scenario here. Bible dominates with a whopping *985920* user reviews! And Dictionary.com is right below it with 200047 user reviews.

So what does this mean? Well, Navigation and Reference apps aren't exactly "fun" apps. In other words, there could be potential for an app designed for practical purpose in the iOS market. However, Navigation has more dominating apps compared to Reference, so it might be easier to slide into that market if we want to develop a new app idea.

Let's move onto Google Play.

#### Google Play

In [25]:
display_table(android_new, 5)

1,000,000+ : 15.726534296028879
100,000+ : 11.552346570397113
10,000,000+ : 10.548285198555957
10,000+ : 10.198555956678701
1,000+ : 8.393501805054152
100+ : 6.915613718411552
5,000,000+ : 6.825361010830325
500,000+ : 5.561823104693141
50,000+ : 4.7721119133574
5,000+ : 4.512635379061372
10+ : 3.5424187725631766
500+ : 3.2490974729241873
50,000,000+ : 2.3014440433213
100,000,000+ : 2.1322202166064983
50+ : 1.917870036101083
5+ : 0.78971119133574
1+ : 0.5076714801444043
500,000,000+ : 0.2707581227436823
1,000,000,000+ : 0.22563176895306858
0+ : 0.04512635379061372
0 : 0.01128158844765343


We already have information on the number of installs, so viewing the data is pretty straightforward. Since we need to calculate the average number of user installs, we'll also need to convert every install number to `float` (commas and plus characters will raise an error).

In [26]:
android_cats = freq_table(android_new, 1)

for c in android_cats:
    total = 0
    cat_len = 0
    for a in android_new:
        cat_app = a[1]
        if cat_app == c:
            a_installs = a[5]
            a_installs = a_installs.replace('+', '')
            a_installs = a_installs.replace(',', '')
            total += float(a_installs)
            cat_len += 1
    avg_num_installs = total / cat_len
    print(c, ': {}'.format(avg_num_installs))

ART_AND_DESIGN : 1986335.0877192982
AUTO_AND_VEHICLES : 647317.8170731707
BEAUTY : 513151.88679245283
BOOKS_AND_REFERENCE : 8767811.894736841
BUSINESS : 1712290.1474201474
COMICS : 817657.2727272727
COMMUNICATION : 38456119.167247385
DATING : 854028.8303030303
EDUCATION : 1833495.145631068
ENTERTAINMENT : 11640705.88235294
EVENTS : 253542.22222222222
FINANCE : 1387692.475609756
FOOD_AND_DRINK : 1924897.7363636363
HEALTH_AND_FITNESS : 4188821.9853479853
HOUSE_AND_HOME : 1331540.5616438356
LIBRARIES_AND_DEMO : 638503.734939759
LIFESTYLE : 1437816.2687861272
GAME : 15588015.603248259
FAMILY : 3695641.8198090694
MEDICAL : 120550.61980830671
SOCIAL : 23253652.127118643
SHOPPING : 7036877.311557789
PHOTOGRAPHY : 17840110.40229885
SPORTS : 3638640.1428571427
TRAVEL_AND_LOCAL : 13984077.710144928
TOOLS : 10801391.298666667
PERSONALIZATION : 5201482.6122448975
PRODUCTIVITY : 16787331.344927534
PARENTING : 542603.6206896552
WEATHER : 5074486.197183099
VIDEO_PLAYERS : 24727872.452830188
NEWS_AND_

We can see that communication apps, on average, have the highest number of installs. Let's take a deeper look into this.

In [27]:
for a in android_new:
    if a[1] == 'COMMUNICATION' and (a[5] == '1,000,000,000+' or a[5] == '500,000,000+'):
        print(a[0], ':', a[5])

WhatsApp Messenger : 1,000,000,000+
Google Duo - High Quality Video Calls : 500,000,000+
Messenger – Text and Video Chat for Free : 1,000,000,000+
imo free video calls and chat : 500,000,000+
Skype - free IM & video calls : 1,000,000,000+
LINE: Free Calls & Messages : 500,000,000+
Google Chrome: Fast & Secure : 1,000,000,000+
UC Browser - Fast Download Private & Secure : 500,000,000+
Gmail : 1,000,000,000+
Hangouts : 1,000,000,000+
Viber Messenger : 500,000,000+


Similar to what we saw in the App Store, there are a few apps that dominate the Communication category. WhatsApp Messenger, Google Duo, Messenger, Skype, Google Chrome, Gmail are a few leading apps. However, since these apps drown out the others, it can largely influence the average number of communication apps installed.

Since we saw potential for the Reference genre in the App Store, I think it would be a good idea to check out that category in Google Play. The Books and Reference category has an average of 8767811 installs, which isn't bad! Let's look into that a bit further.

In [28]:
for a in android_new:
    if a[1] == 'BOOKS_AND_REFERENCE' and (a[5] == '1,000,000,000+' or a[5] == '500,000,000+'or a[5] == '100,000,000+'):
        print(a[0], ':', a[5])

Google Play Books : 1,000,000,000+
Bible : 100,000,000+
Amazon Kindle : 100,000,000+
Wattpad 📖 Free Books : 100,000,000+
Audiobooks from Audible : 100,000,000+


We can see there are only a few leading apps here, so there might be a promising market for an app idea.

In [29]:
for a in android_new:
    if a[1] == 'BOOKS_AND_REFERENCE' and (a[5] == '1,000,000+' or a[5] == '50,000,000+'or a[5] == '10,000,000+'):
        print(a[0], ':', a[5])

Wikipedia : 10,000,000+
Cool Reader : 10,000,000+
Book store : 1,000,000+
FBReader: Favorite Book Reader : 10,000,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
ReadEra – free ebook reader : 1,000,000+
eBoox: book reader fb2 epub zip : 1,000,000+
All Maths Formulas : 1,000,000+
HTC Help : 10,000,000+
Moon+ Reader : 10,000,000+
English-Myanmar Dictionary : 1,000,000+
Golden Dictionary (EN-AR) : 1,000,000+
All Language Translator Free : 1,000,000+
Aldiko Book Reader : 10,000,000+
Al-Quran (Free) : 10,000,000+
Al Quran Indonesia : 10,000,000+
Al'Quran Bahasa Indonesia : 10,000,000+
Al Quran Al karim : 1,000,000+
Koran Read &MP3 30 Juz Offline : 1,000,000+
Hafizi Quran 15 lines per page : 1,000,000+
Quran for Android : 10,000,000+
Satellite AR : 1,000,000+
Oxford A-Z of English Usage : 1,000,000+
Dictionary.com: Find Definitions for English Words : 10,000,000+
English Dictionary - Offline : 10,000,000+
NOOK: Read eBoo

There's a wide variety of plenty e-books, quote apps, cloud-based softwares for books, and religious texts. So, this might serve as a good starting point for an app idea. There are only two dominating Reference apps in the App Store and five in Google Play. Compared to other genres and categories, this allows some room in the market for a new and hopefully profitable app!
***

## Conclusion

Overall, we were able to analyze data about the App Store and Google play to recommend a profitable app in both markets for developers. Since we were working with free apps, we targeted in-app ads as our main source of revenue and dug deeper into customer engagement across different app profiles.

In our results, we were able to conclude that although the market seems a bit intimidating due to some tech giants, there could be some room for a new app idea which would possibly include e-reading, quotes, dictionary references, or religious text.